src/main/java/com/epam/poc/commons/BasePage.java

```diff
@@ -193,4 +199,62 @@ public void waitForPageTitleAsExpectedText(WebDriver driver, String pageTitleVal
             logger.info("Page title is not the same as expected text");
         }
     }
+
+    @Step("Get a new tab url")
+    public String getNewTabUrl(WebDriver driver) {
+        Set<String> handles = driver.getWindowHandles();
+        for (String actual : handles) {
+            if(!actual.equalsIgnoreCase(this.getCurrentUrl(driver))){
+                driver.switchTo().window(actual);
+                waitForPageLoadedCompletely(driver);
+            }
+        }
+        return driver.getCurrentUrl();
+    }
+
+    @Step("Hover on element {1}")
+    public <Page extends BasePage> Page hover(WebDriver driver, WebElement webElement){
+        Actions actions = new Actions(driver);
+        actions.moveToElement(webElement).build().perform();
+        return (Page) this;
+    }
+
+
+    public <Page extends BasePage> Page display(WebElement webElement){
```

Comment:
You should avoid Assert in the BasePage class, just use in Test class. So let's remove all methods that contains "Assert"